src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/TextEditingTargetNotebook.java

```diff
@@ -1490,6 +1572,75 @@ public void run()
       }
    };
    
+   private int getPlotWidth()
+   {
+      // subtract some space to account for padding; ensure the plot doesn't
+      // grow arbitrarily large. note that this value must total the amount of
+      // space outside the element (here, 2 * (10px margin + 1px border)); since
+      // we stretch the plot to fit the space it will scale in unpredictable
+      // ways if it doesn't fit exactly
+      return Math.min(docDisplay_.getPixelWidth() - 22,
+                      ChunkOutputUi.MAX_PLOT_WIDTH);
+   }
+   
+   private Timer resizePlotsLocal_ = new Timer()
+   {
+      @Override
+      public void run()
+      {
+         for (ChunkOutputUi output: outputs_.values())
+         {
+            output.getOutputWidget().syncHeight(false, false);
+         }
+      }
+   };
+   
+   private Timer resizePlotsRemote_ = new Timer()
```

Comment:
To further underline my previous comment, I'm thinking that if it takes e.g. 3 seconds for the user to resize that there should be one resize event that happens at the end.